my-app/src/ListIterator.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class ListIterator extends Component {
+  constructor(props) {
+    super(props);
+    this.handleDelete = this.handleDelete.bind(this);
+  }
+
+  handleDelete(event) {
+    let clickedButtonElement = event.currentTarget.previousElementSibling.textContent;
```

Comment:
instead of doing this, take a parameter either index or item from where you are calling this function. 
The way in which you are doing isn't the correct way.